### DISCUSSION TITLE
Danfoss Ally temperature calibration

### DISCUSSION THREAD

**Discussion by ipeacocks** [Koenkk/zigbee2mqtt#10130] [Devices]
I have 3 Danfoss Ally TRVs and in general happy with them. I only don't like one thing there - absence of temperature calibration. I've heard that there is `External_measured_room_sensor` which can influence on current temp parameter. It's said that it counts average temperature and use it for work.

I've tried to change it in zigbee2mqtt but I can't see any difference in `local_temperature` parameter. Am I getting logic correct?
Can somebody share own experience in calibrating temp of this TRV?

**Comment by Char-r** [comment]
It is not expposed in Z2M but it seems to exist in the documentation:

https://raw.githubusercontent.com/SimpleUserHA/DanfosseTRV/main/Danfoss%20Ally%20Radiator%20Thermostat%201.18%20Zigbee%20Cluster%20Specifications%2001122021-AM375549618098en-000102.pdf

Regulation SetPoint Offset Int8

in steps of 0.1°C.
The range of this offset is –2.5 °C to +2.5 °C (0xE7 ...
0x19).

**Reply by 2Max2Pax** [reply]
I tryed reading the attribute danfossRegulationSetpointOffset:
![danfossRegulationSetpointOffset](https://user-images.githubusercontent.com/86773160/149766709-3bb75496-25da-4ed3-8442-eef325addb7c.png)
I got the following error:
```
Publish 'set' 'read' to 'trv_studio' failed: 'Error: Read 0x804b50fffe3da0a8/1 hvacThermostat(["danfossRegulationSetpointOffset"], {"sendWhenActive":true,"timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":null,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Status 'UNSUPPORTED_ATTRIBUTE')'
```
